Src/Test.UnitTests.Sarif.PatternMatcher.Cli/AnalyzeCommandTests.cs

```diff
@@ -17,6 +19,27 @@ namespace Microsoft.CodeAnalysis.Sarif.PatternMatcher.Cli
 {
     public class AnalyzeCommandTests
     {
+        [Fact]
+        public void AnalyzeCommand_DefinitionsArgumentIsRequired()
+        {
+            string[] args = new[]
+            {
+                    "analyze",
+                    Guid.NewGuid().ToString(),
+                    $"-o", Guid.NewGuid().ToString(),
+            };
+
+            Program.ClearUnitTestData();
+            int result = Program.Main(args);
+            result.Should().Be(CommandBase.FAILURE);
+
+            // This validation is sufficient because the null check for an
+            // instantiated analyze command verifies that we failed the 
+            // CommandLine parsing code and error out before attempting
+            // analysis.
+            Program.InstantiatedAnalyzeCommand.Should().BeNull();
+        }
```

Comment:
Did this test fail before the change?

Using `spam analyze test.txt -o test2.txt` here are the before and after results:
* before: ```Analyzing...
SPAM : error ERR997.AllRulesExplicitlyDisabled

Analysis did not complete due to one or more unrecoverable execution conditions.
Unexpected fatal runtime condition(s) observed: NoRulesLoaded```
* after: ```
Analyzing...
SPAM : error ERR997.AllRulesExplicitlyDisabled : All rules were explicitly disabled so there is no work to do.

Analysis did not complete due to one or more unrecoverable execution conditions.
Unexpected fatal runtime condition(s) observed: NoRulesLoaded```

Reply:
Can you run your test again after syncing? I really appreciate this in-depth examination, it's super helpful.

What you should see is that CommandLine lib parsing should error out, w/something like so.

I have to check in to unblock other work, but please do comment if I've missed something and I'll fold changes into other work lighting up today.

ERROR(S):
  Required option 'd, search-definitions' is missing.

Reply:
Well, this is interesting. Two scenarios get the expected result and two get the previous result. Unit test and running via debugger: exactly as expected. Use the command line (same debug output folder debugger runs off of or release): get the previous result.